frontend/src/components/rootModal/index.tsx

```diff
@@ -4,41 +4,46 @@ import { useDispatch, useSelector } from 'react-redux';
 
 import {
   hideModal,
-  ModalState,
+  ModalConfiguarion,
   ModalType,
-  selectModalDetails,
-} from '../../features/modalSlice';
+  selectModalConfiguration,
+} from '../../store/modalSlice';
 import CreateUserModal from '../createUserModal';
 import DeleteUserModal from '../deleteUserModal';
+import { ModalProps } from './modalProps';
 import rootModalSyles from './rootModal.module.scss';
 
-type ModalHash = {
-  [key in ModalType]: (arg0: any) => JSX.Element;
-};
+/* eslint-disable react/jsx-props-no-spreading */
+const getModal = (modalState: ModalConfiguarion, defaultProps: ModalProps) => {
+  const modalType = modalState.type;
 
-const MODAL_COMPONENTS: ModalHash = {
-  DELETE_USER: DeleteUserModal,
-  CREATE_USER: CreateUserModal,
+  switch (modalState.type) {
+    case ModalType.DeleteUser: {
+      const { type, ...rest } = modalState;
```

Comment:
Minor nitpick, I believe this line can be pulled out of the switch statement (can replace `const modalType = modalState.type`)

Reply:
Afraid its not possible https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions

If we destructure the `type` & `rest` outside the switch the variables become disassociated and we don't know what type `rest` is. 

The reason for `modalType` var is because the default case is strictly currently never possible and TS knows this. I added the default case for when we add a new `ModalType` and forget to add case to the switch statement.